src/core/actions/unitactions/Capture.java

```diff
@@ -23,20 +24,38 @@ public City getTargetCity() {
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute Capture city actions
-        return null;
+
+        LinkedList<Action> captures = new LinkedList<>();
+        if(isFeasible(gs)){
```

Comment:
This wouldn't work either, targetCity wouldn't be set here. We need to return all the possible capture actions this unit can make. In essence, this will be one action (or none) if the unit is at a tile with a city or a village that doesn't belong to its tribe.